app/cronjob/trailingTrade/step/handle-open-orders.js

```diff
@@ -88,7 +55,23 @@ const execute = async (logger, rawData) => {
     }
     // Is the stop price is higher than current limit price?
     if (order.side.toLowerCase() === 'buy') {
-      if (parseFloat(order.stopPrice) >= buyLimitPrice) {
+      if (await isExceedingMaxOpenTrades(logger, data)) {
+        // Cancel the initial buy order if max. open trades exceeded
+        data.action = 'buy-order-cancelled';
+        logger.info(
+          { data, saveLog: true },
+          `The current number of open trades has reached the maximum number of open trades. ` +
+            `The buy order will be cancelled.`
+        );
+
+        // Cancel current order
+        const cancelResult = await cancelOrder(logger, symbol, order);
+
+        // Reset buy open orders
+        if (cancelResult === true) {
+          data.buy.openOrders = [];
```

Comment:
Cancelling the order often fails even if it was successful. 

And when it succeeds to cancel, it should process the same as stop > limit price cancel process.

https://github.com/chrisleekr/binance-trading-bot/pull/462/files#diff-48b71d5ac0a2a4fd94708519e33e808e09b1a98ab8df1b8996f499d8b71a244bR124

Agree? @uhliksk @habibalkhabbaz

Reply:
I agree.

Reply:
Let me quickly refactor it.

Reply:
I agree. It'll be cleaner way even as this is actually cancelling the first buy order which have no impact on anything. It will save some headache in future if it will be reused for another purpose.

Reply:
I was gonna just update this file, but @uhliksk is right. I will make a generic function in the common.js and update other steps to reduce duplicated codes.

Allow me some time.

Reply:
@uhliksk @habibalkhabbaz 

I've added a generic function `refreshOpenOrdersAndAccountInfo ` to `common.js` - https://github.com/chrisleekr/binance-trading-bot/pull/462/files#diff-478c565f3dc02bbdb099f674677ebcd7d1f16fa4ddd06220dd3d6119b8467b84R1130

And I updated `handle-open-orders.js` to use  `refreshOpenOrdersAndAccountInfo` - https://github.com/chrisleekr/binance-trading-bot/pull/462/files#diff-48b71d5ac0a2a4fd94708519e33e808e09b1a98ab8df1b8996f499d8b71a244bR127

I checked and found some other steps have exactly the same code.
However, refactoring those steps seems out of scope. Hence, I've added FIXME comments instead of refactoring them - https://github.com/chrisleekr/binance-trading-bot/pull/462/files#diff-ad373fc5bc2ac395a075a4ef44bdaebd17d66fc02707c5c07c83e1483199bab6R205

Let's test it out for another day and if there is no issue, I will merge in.

This is a very good fix @uhliksk @habibalkhabbaz